File: 2298-count-integers-with-even-digit-sum/count-integers-with-even-digit-sum.js
var countEven = function (num) {
    let dSum = 0, temp = num
    while (temp > 0) {
        dSum += temp % 10
        temp = Math.floor(temp / 10)
    }
    return (dSum % 2 === 0) ? Math.floor(num / 2) : Math.floor((num - 1) / 2)
};

File: 2298-count-integers-with-even-digit-sum/count-integers-with-even-digit-sum.py
class Solution:
    def countEven(self, num: int) -> int:
        # Calculate digit sum of the boundary 'num'
        digit_sum = sum(int(d) for d in str(num))
        
        # If the boundary's digit sum is even, it 'completes' a pair
        if digit_sum % 2 == 0:
            return num // 2
        else:
            return (num - 1) // 2